### DISCUSSION TITLE
New parser & remark-frontmatter

### DISCUSSION THREAD

**Discussion by mrzmmr** [remarkjs/remark#538] [Q&A] [answered]
I'm not sure if this is a bug or being worked out but I can't seem to get `remark-frontmatter` >= 3 to work with `remark@next`. I've tried using with and without options but yaml is still parsed as a `thematicBreak`. If anyone has any insight it'd be appreciated :)

**package.json**
```js
{
    "dependencies": {
        "remark": "^13.0.0-alpha.1",
        "remark-frontmatter": "^3.0.0"
    }
}
```

***index.js***
```js
let remark = require('remark')
let frontmatter = require('remark-frontmatter')
let md = `
---
title: test
---

# Test
`

let output = remark()
    .use(frontmatter, ['yaml'])
    .parse(md)

console.dir(output)
```

***output***
```text
{
  type: 'root',
  children: [
    { type: 'thematicBreak', position: [Object] },
    {
      type: 'heading',
      depth: 2,
      children: [Array],
      position: [Object]
    },
    {
      type: 'heading',
      depth: 1,
      children: [Array],
      position: [Object]
    }
  ],
  position: {
    start: { line: 1, column: 1, offset: 0 },
    end: { line: 7, column: 1, offset: 29 }
  }
}
```

**Comment by wooorm** [comment, marked as answer]
This indeed changed. Frontmatter still works, but must start on the first line:

```diff
-let md = `
----
+let md = `---
 title: test
 ---

 # Test
 `
```

---

> * Frontmatter must now start on the first line
>   Previously, delaying with blank lines first worked.
>   That behavior does not match how GitHub parses YAML, and is thus removed.
>   — [source](https://github.com/remarkjs/remark-frontmatter/commit/fb1357b0ece628ec9ede5fdeed77463fe2661e9a)

**Comment by mrzmmr** [comment]
Oops ok makes sense thanks!
